co-noir/co-acvm/src/solver/memory_solver.rs

```diff
@@ -0,0 +1,110 @@
+use acir::{
+    acir_field::GenericFieldElement,
+    circuit::opcodes::{BlockId, MemOp},
+    native_types::{Expression, Witness},
+};
+use ark_ff::PrimeField;
+use mpc_core::traits::NoirWitnessExtensionProtocol;
+
+use super::{CoAcvmResult, CoSolver};
+
+impl<T, F> CoSolver<T, F>
+where
+    T: NoirWitnessExtensionProtocol<F>,
+    F: PrimeField,
+{
+    pub(super) fn solve_memory_init_block(
+        &mut self,
+        block_id: BlockId,
+        init: &[Witness],
+    ) -> CoAcvmResult<()> {
+        tracing::trace!("solving memory init block {}", block_id.0);
+        // TODO: should we trust on the compiler here?
```

Comment:
I would say that we can leave this there as inits should not dominate the runtime anyways